plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/AwsSigner.java

```diff
@@ -34,20 +34,29 @@ private AwsSigner() {
      * @param configuration AWS Client configuration
      * @throws IllegalArgumentException if signer does not exist
      */
-    public static void configureSigner(String signer, ClientConfiguration configuration)
+    public static void configureSigner(String signer, ClientConfiguration configuration, String endpoint)
         throws IllegalArgumentException {
 
         if (signer == null) {
             throw new IllegalArgumentException("[null] signer set");
         }
 
-        try {
-            // We check this signer actually exists in AWS SDK
-            // It throws a IllegalArgumentException if not found
-            SignerFactory.getSignerByTypeAndService(signer, null);
-            configuration.setSignerOverride(signer);
-        } catch (IllegalArgumentException e) {
-            throw new IllegalArgumentException("wrong signer set [" + signer + "]");
+        if (signer.equals("S3SignerType")) {
+            // V2 signer type is not support in aws China and Frankfurt region
+            if (endpoint.equals("s3.cn-north-1.amazonaws.com.cn") ||
+                    endpoint.equals("s3.eu-central-1.amazonaws.com")) {
+                throw new IllegalArgumentException(endpoint + " does not support S3SignerType");
```

Comment:
I think we should only log.warn here. I can imagine that at some point AWS may support this in China and I would not block users for this.
May be we should not control that at all and let the user specify whatever he wants.

I mean that if this plugin is used with other S3 compatible platform, we can't do those checks.